### ISSUE TITLE
torch.cat behaves differently on Tensor vs Variable (also a backward compatibility issue)

### ISSUE THREAD

**Issue by B1azingB1ade** [pytorch/pytorch#5402]
- OS: Ubuntu 16.04.3 LTS
- PyTorch version: 0.3.1
- How you installed PyTorch (conda, pip, source): pip
- Python version: 2.7
- CUDA/cuDNN version: 9.1

I wanted to concatenate a single variable along a given dimension. I was able to do it for PyTorch 0.2, but 0.3 would give me the error:

> RuntimeError: cat(): argument 'tensors' (position 1) must be tuple of Variables, not Variable

However, I'd like to be able to concatenate a Variable along a dimension instead of creating a tuple of Variables. When the input is a Tensor instead of a Variable, torch.cat() still yields the desired output. A script to reproduce the error is as follows:

```
import torch
import numpy as np
from torch.autograd import Variable
x = torch.Tensor(np.array(range(24)).reshape(2, 3, 4))

print torch.cat(x, dim=1)
# No issue here. Gives me the desired concatenation.

print torch.cat(Variable(x), dim=1)
# Error message complains about input being a single Variable.
```
In short, torch.cat() can concatenate a single Variable along a given dimension in 0.2 but not in 0.3, while torch.cat() can concatenate a single Tensor along a given dimension in both versions.

**Comment by zou3519** [comment]
@ailzhang

**Comment by ailzhang** [comment]
I can repro this. Working on a fix.

**Comment by ailzhang** [comment]
Hey @B1azingB1ade, actually after the Tensor/Variable merge they both throw this error in pytorch master. Discussed with @colesbury and we would like to keep it requiring tuple of Tensors since that's closer to the definition of 'concatenate'.
The cat on single Tensor can easily be done by view/reshape. Or if you would like to use torch.cat, torch.cat(list(x), dim=1) also works.

**Comment by B1azingB1ade** [comment]
list(x) works. I've tested that the trained model yielded expected results, too. Thanks!
